accepted/2021/Globalization/GlobalizationSize.md

```diff
@@ -0,0 +1,235 @@
+
+In .NET 5.0 we went one step further to [support using ICU library](https://docs.microsoft.com/en-us/dotnet/standard/globalization-localization/globalization-icu) when running on Windows and we made this behavior the default one. Also, we have supported the [ICU app-local](https://docs.microsoft.com/en-us/dotnet/standard/globalization-localization/globalization-icu#app-local-icu) feature which allows applications to use a different ICU version than the global system one if needed.
+
+In .NET Core, we have supported the [Globalization Invariant Mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md) is the mode that the application which doesn't care much about globalization can be configured with mode. The benefit of enabling this mode is avoiding any dependencies on the system or the ICU. The Invariant mode is mainly introduced to support smaller container images (e.g. Alpine Linux image). Another benefit of the Invariant Mode is guaranteeing consistent behavior of the app when running on different OS's or platforms.
+
+### Concerned Scenarios
+
+Although globalization support is working fine for desktop and server applications, it was concerning in other platforms mainly because it required installing ICU library. The ICU module sizes are concerning and wouldn't be acceptable especially when running on mobile platforms.
+
+```
+ICU module sizes from MS-ICU Linux x64 package:
+
+libicudata.so.68.2.0.6    28,320,992
+libicui18n.so.68.2.0.6    4,220,936
+libicuuc.so.68.2.0.6      2,479,488
+
+Thanks to @filipnavara and @CoffeeFlux providing the following info:
+
+iOS arm64:
+icudt.dat 1,713,152
+icudt_CJK.dat 1,006,224
+icudt_EFIGS.dat 602,096
+icudt_no_CJK.dat 1,271,296
+libicudata.a 720
+libicui18n.a 3,536,472
+libicuuc.a 2,286,376
+
+Android arm64:
+icudt.dat 1,512,896
+icudt_CJK.dat 966,080
+icudt_EFIGS.dat 559,568
+icudt_no_CJK.dat 1,082,128
+libicudata.a 1,116
+libicui18n.a 7,105,274
+libicuuc.a 4,267,870
+
+Browser WASM:
+icudt.dat 1,512,896
+icudt_CJK.dat 966,080
+icudt_EFIGS.dat 559,568
+icudt_no_CJK.dat 1,082,128
+libicui18n.a 4,077,896
+libicuuc.a 2,510,674
+
+The native ICU bits end up being about 380KB compressed on wasm.
+
+```
+
+- **Mobile Platforms**:
+  - **Android**: Although Android OS comes with the ICU libraries which .NET can use, the OEMs can choose to not include such libraries in their device images.
+  - **iOS/MacCatalyst**: don't come with an ICU library that .NET can use. .NET had to provide an ICU package to be used in such platforms.
+- **Web assembly**: The clients have to be a small size and need to run inside the browser which will not allow accessing any libraries outside the browser. That means WebAssembly clients need to include an ICU package for globalization support.
```

Comment:
To what extent do browsers provide globalization functionality that we could leverage without having to include it in the app?

Reply:
@lewing you may have some more info about that as you have investigated such options before. could you please share what you got during then?